mappings/net/minecraft/server/world/SleepManager.mapping

```diff
@@ -0,0 +1,18 @@
+CLASS net/minecraft/class_5838 net/minecraft/server/world/SleepManager
+	FIELD field_28866 total I
+	FIELD field_28867 sleeping I
+	METHOD method_33811 clearSleeping ()V
+	METHOD method_33812 canSkipNight (I)Z
+		ARG 1 percentage
+	METHOD method_33813 isLongEnough (ILjava/util/List;)Z
```

Comment:
I would recommend changing this name along with the one in `PlayerEntity` to better indicate that this is based off the player's sleep timer. Would something like `PlayerEntity#hasFullySlept` and `SleepManager#haveAllFullySlept` work?

Reply:
Is there one in player entity? The whole check is done in this method I recall

Reply:
```java
public boolean isLongEnough(int percentage, List<ServerPlayerEntity> players) {
	int sleepingLongEnoughCount = players.stream().filter(PlayerEntity::isSleepingLongEnough).count();
	return sleepingLongEnoughCount >= this.getNightSkippingRequirement(percentage);
}
```

Reply:
Maybe `hasSleptLongEnough`. If you make a pr, I will endorse it

Reply:
Decided to call this `canResetTime` vs the plain `canSkipNight`